### ISSUE TITLE
Hidden stories still notify about comments

### ISSUE THREAD

**Issue by pushcx** [lobsters/lobsters#947]
The `replying_comments` view needs to be updated to ignore replies if the user has hidden the story.

`Comment#deliver_mention_notifications` needs to ignore replies if the receiver has hidden the story.

`Comment#deliver_reply_notifications` needs to ignore replies if the receiver has hidden the story. Maybe this is best implemented in `users_following_thread` but I haven't audited all functions that call it.

**Comment by Jaskaran2** [comment]
Hi is this issue still up for grabs??

**Comment by pushcx** [comment]
Yup. Issues tagged 'good first issue' are pretty much always ready to be picked up! We don't assign issues and I try to only rarely (and explicitly) claim issues myself.
